IMC_WhyReact/blog.md

```diff
@@ -0,0 +1,239 @@
+
+1. React class component
+```js
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+// class
+class NewComponent extends React.Component {
+  render() {
+    return (
+      <div>hello</div>
+    )
+  };
+};
+
+ReactDOM.render(
+  <NewComponent />,
+  document.getElementById('root')
+)
+```
+
+
+2. React functional component (another way to write the above code)
+```js
+// function
+function AnotherComponent() {
+  return (
+    <div>hello</div>
+  );
+};
+
+ReactDOM.render(
+  <AnotherComponent />,
+  document.getElementById('root')
+)
+```
+
+
+3. React JSX
+```js
+// jsx
+let anime = 'Hunter x Hunter';
+let element = <div>My favorite anime is {anime}.</div>;
+ReactDOM.render(
+  element,
+  document.getElementById('root')
+);
+```
+
+4. React inline CSS 
+```js
+// inline css as an object
+let element = 
+  <div 
+    style={{border:'solid purple 2px', padding:'5em'}}>
+    My favorite anime is {anime}
+  </div>
+ReactDOM.render(
+  element,
+  document.getElementById('root')
+);
+```
+
+5. React component with state
+```js
+// state
+class ClassComponentWithState extends React.Component {
+  constructor() {
+    this.state = {
+      animes: [
+        'Hunter x Hunter',
+        'One Piece',
+        'Neon Genesis Evangelion'
+      ],
+      recs: []
+    };
+    this.handleClick = this.handleClick.bind(this)
+  };
+
+  handleClick() {
+    if (this.state.animes.length !== 0) {
+      this.setState(state => ({
+        animes: state.animes.slice(0, state.animes.length-1),
+        recs: state.recs.push(state.animes[state.animes.length-1])
+      }))
+    }
+
+    else {
+      alert('We have no more anime recommendations for you today! Please return after you have finished watching the ones we have already given you. Good luck!')
+    }
+  }
+
+  render() {
+    let recommended = recs.map(anime => <p style={{margin:'.5em'}}>{anime}</p>)
+
+    return (
+      <div style={{display:'block', padding:'1.5em'}}>
+        <button 
+          style={{padding:'.5em', margin:'1em'}}
+          onClick={this.handleClick}>
+          Click to get new anime recommendation!
+        </button>
+        {recommended}
+      </div>
+    );
+  };
+};
+
+ReactDOM.render(
+  <ClassComponentWithState />,
+  document.getElementById('root')
+);
+```
+
+6. React component with props
+```js
+// props
+let FunctionalComponentWithProps = (props) => {
+  return (
+    <div style={props.styles}>
+      <h1>You should watch the following animes!</h1>
+      {props.animes.map(anime => <p>{anime}</p>)}
+    </div>
+  )
+}
+
+let styles = {
+  display: 'block',
+}
+
+let animes = [
+  'Hunter x Hunter',
+  'One Piece',
+  'Neon Genesis Evangelion'
+]
+
+ReactDOM.render(
+  <FunctionalComponentWithProps 
+    styles={styles}
+    animes={animes}
+  />,
+  document.getElementById('root')
+);
+```
+
```

Comment:
Very useful code snippets! However, I feel like you just dump them on the reader without giving enough detail about the code snippets can actually be _used_ in an application. Either provide examples and elaborate more in this section, or get rid of this section and integrate these snippets + examples into your two above sections.